### PR TITLE
[Feature:Developer] Add CI Mode to Config

### PR DESCRIPTION
**Pull request by IDzyre**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
#10097 has a function that generates a random code that is sent via an email to the user to verify their email. Cypress does not have access to the emails, which makes it hard to test.

### What is the new behavior?
A flag has been added to the Submitty config to tell the system it is in CI. This allows the functions to use a preset code instead of a random code, which then allows for testing of the verification part of the system. The actual sending of the emails is tested by the PHP email unit testing.

### CODE REVIEW

File: migration/migrator/migrations/system/20240828152944_ci_mode.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    """
+    edited_config = config.submitty
+    if ('is_ci' not in edited_config):
+        edited_config['is_ci'] = False
+    
+    with open(config.config_path / 'submitty.json', 'w') as file_path:
+        json.dump(edited_config, file_path, indent=4)
```
Comment (williamjallen) [inline migration/migrator/migrations/system/20240828152944_ci_mode.py:16]: I don't like the idea of adding a field to `submitty.json` purely for testing purposes.  Is there a practical difference between this and the existing debug field?

File: site/app/models/Config.php [hunk 2]
```diff
@@ -167,6 +167,12 @@ class Config extends AbstractModel {
      * @var string */
     protected $cgi_tmp_path;
 
+    /**
+     * Indicates whether Submitty is running in CI (to mock certain 'random' functions)
+     * @var bool
+     * @prop
+     */
+    protected $ci = false;
```
Comment (williamjallen) [inline site/app/models/Config.php:175]: Isn't this the same as `$debug` in practice?

**Review by williamjallen — CHANGES_REQUESTED** [review]

File: site/app/models/Config.php (hunk 2 above)
Comment (IDzyre) [reply site/app/models/Config.php:175]: I guess it is, from reading the phpdoc about debugging, I assumed that it was mainly for development environments to output full exceptions, but I guess we can use it as a 'ci mode' as well..

**Comment by MasterOdin** [comment]
Depending on how/what your testing, [`phpmock`](https://github.com/php-mock/php-mock) may work for your needs as well. It's already used in a number of places to shim over builtin PHP functions as part of unit tests.

**Pull request closed without merging**